mathesar_ui/src/sections/import-data/importUtils.ts

```diff
@@ -378,6 +378,31 @@ export function cancelImport(fileImportStore: FileImport): void {
   }
 }
 
+export async function importFromURL(fileImportStore: FileImport, url: string): Promise<void> {
+  setInFileStore(fileImportStore, {
+    previewTableCreationStatus: States.Loading,
+  });
```

Comment:
@pavish This works in disabling the button and showing the spinner when an user clicks on it. However, we also call this in the `createPreviewTable` function nested inside the `loadPreview` function. It works fine, but seems wrong to be calling it twice.

Reply:
@mr-gabe49 This is because previously we needed to show the loading indicator only when during the preview step. In this case, we need to show it based on `uploadStatus` as well.

Instead of setting `previewTableCreationStatus` in this section, we should be setting `uploadStatus`